src/sonic-yang-models/sonic-vlan.yang

```diff
@@ -0,0 +1,186 @@
+                            error-message "Invalid Vlan name pattern";
+                            error-app-tag vlan-name-invalid;
+                        }
+                    }
+                }
+
+                leaf vlanid {
+                    mandatory true;
+                    type uint16 {
+                        range "1..4095" {
+                            error-message "Vlan ID out of range";
+                            error-app-tag vlanid-invalid;
+                        }
+                    }
+                }
+
+                leaf-list members {
+                    must "count(current()/../members[text()=/spc:sonic-portchannel/spc:PORTCHANNEL_MEMBER/" +
+                        "spc:PORTCHANNEL_MEMBER_LIST[spc:ifname=current()]/spc:name]) = 0 and " +
+                        "count(current()/../members[text()=/spc:sonic-portchannel/spc:PORTCHANNEL_MEMBER/" +
+                        "spc:PORTCHANNEL_MEMBER_LIST[spc:name=current()]/spc:ifname]) = 0 " {
+                            error-message "A vlan interface member cannot be part of portchannel which is already a vlan member";
+                        }
+
+
+                    type union {
+                        type leafref {
+                            path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:ifname";
+                        }
+                        type leafref {
+                            path "/spc:sonic-portchannel/spc:PORTCHANNEL/spc:PORTCHANNEL_LIST/spc:name";
+                        }
+                    }
+                }
+                leaf mtu  {
+                    type uint32 {
+                        range "1312..9216" {
+                            error-message "Invalid MTU value";
+                            error-app-tag mtu-invalid;
+                        }
+                    }
+                }
+
+                leaf admin_status {
+                    type scommon:admin-status;
+                }
+            }
+        }
+
+        container VLAN_MEMBER { 
+
+            list VLAN_MEMBER_LIST { 
+                key "name ifname";
+
+                leaf name {
+                    type leafref {
+                        path "../../../VLAN/VLAN_LIST/name";
+                    }
+                }
+
+                leaf ifname {
+                    type union {
+                        type leafref {
+                            path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:ifname";
+                        }
+                        type leafref {
+                            path "/spc:sonic-portchannel/spc:PORTCHANNEL/spc:PORTCHANNEL_LIST/spc:name";
+                        }
+                    }
+                }
+
+                leaf tagging_mode {
+                    type scommon:tagging_mode;
+                    default tagged;
+                }
+            }
+        }
+
+
+        container VLAN_TABLE {
+            config false;
```

Comment:
Which DB is this from?

Reply:
Yeah, same ques. We need VLAN VRF entry as below [I think, as per VRF HLD]. But VLAN_table, I am not sure.

   "VLAN_INTERFACE": {
        "Vlan100": {},
        "Vlan777": {},
        "Vlan100|2a04:....::1/64": {
            "family": "IPv6", 
            "scope": "global"
        }, 
        "Vlan100|10...../26": {
            "family": "IPv4", 
            "scope": "global"
        }, 
        "Vlan100|fe80::..../10": {
            "family": "IPv6", 
            "scope": "local"
        }, 
        "Vlan777|2a04:..../64": {
            "family": "IPv6", 
            "scope": "global"
        },

Reply:
This is not Vlan table. Its similar to LAG/PORT Tables. L3 is binded to another table which is suffex-ed with _INTERFACE which has VRF info. In the current context, I think it is the VLAN table that's referenced here. @renukamanavalan , its in APPL_DB